npm_and_yarn/lib/dependabot/npm_and_yarn/version.rb

```diff
@@ -62,6 +62,7 @@ def self.semver_for(version)
 
       sig { override.params(version: VersionParameter).void }
       def initialize(version)
+        version = clean_version(version)
         @version_string = T.let(version.to_s, String)
         version = version.gsub(/^v/, "") if version.is_a?(String)
```

Comment:
line 67 is also cleaning a version string; can we consolidate?

Reply:
it is also moved.